tests/flows/test_index.py

```diff
@@ -321,6 +324,123 @@ async def test_run_partial_updates_of_concepts_for_document_passages(
     assert test_counts_serialised == json.loads(result["Body"].read().decode("utf-8"))
 
 
+@pytest.mark.asyncio
+@pytest.mark.vespa
+@pytest.mark.slow
+async def test_run_partial_updates_of_concepts_for_document_passages_with_paging(
+    local_vespa_search_adapter: VespaSearchAdapter,
+    vespa_app,
+    mock_bucket,
+    mock_s3_client,
+) -> None:
+    """Test that we can run partial updates of concepts for document passages."""
+    document_import_id = "UNFCCC.party.309.0"
+    document_object_uri = f"s3://{mock_bucket}/labelled_passages/Q218/v4/UNFCCC.party.309.0_translated_en.json"
+
+    # Setup S3
+    file_name = "labelled_passages/Q218/v4/UNFCCC.party.309.0_translated_en.json"
+    data = load_fixture(file_name)
+    body = BytesIO(data.encode("utf-8"))
+    key = "labelled_passages/Q218/v4/UNFCCC.party.309.0_translated_en.json"
+    mock_s3_client.put_object(
+        Bucket=mock_bucket, Key=key, Body=body, ContentType="application/json"
+    )
+
+    # Setup Vespa
+
+    subprocess.run(
+        [
+            "vespa",
+            "feed",
+            "tests/local_vespa/test_documents/family_document_UNFCCC.party.309.0.json",
```

Comment:
Thought: some of this could go in fixtures for reuse ability.

Reply:
Probably![^1]

[^1]: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)